app/controllers/social_networking/profile_pages_controller.rb

```diff
@@ -15,6 +15,8 @@ def show
       if params[:id].present?
         @profile = Profile.find(params[:id])
         store_nudge_initiators(@profile.participant_id)
+      elsif current_participant.social_networking_profile
+        @profile = current_participant.social_networking_profile
       else
         @profile = Profile
                    .find_or_create_by(participant_id: current_participant.id,
```

Comment:
I'd argue for removing the `rescue` and `logger` statement below and letting Sentry report the exception instead.